tests/gacha_tests.cpp

```diff
@@ -0,0 +1,162 @@
+#include "account_tester.hpp"
+#include "variant.hpp"
+
+struct grade {
+   asset                    reward;
+   uint32_t                 score;
+   fc::optional<uint32_t>  limit;
+};
+
+FC_REFLECT(grade, (reward)(score)(limit))
+
+const static name gacha_account_name = N(gxc.gacha);
+
+class gxc_gacha_tester : public gxc_account_tester {
+public:
+   gxc_gacha_tester() {
+      create_accounts({ gacha_account_name });
+      produce_blocks(1);
+
+      _set_code(gacha_account_name, contracts::gacha_wasm());
+      _set_abi(gacha_account_name, contracts::gacha_abi().data());
+      produce_blocks(1);
+
+      auto accnt = control->db().get<account_object,by_name>(gacha_account_name);
+      abi_def abi;
+      BOOST_REQUIRE_EQUAL(abi_serializer::to_abi(accnt.abi, abi), true);
+      abi_ser[gacha_account_name].set_abi(abi, abi_serializer_max_time);
+
+   }
+
+   void subopen(std::array<name,2> scheme, extended_asset budget) {
+      gxc_token_tester::mint(budget);
+      gxc_account_tester::setpartner(scheme.at(1));
+      gxc_token_tester::transfer(N(gxc.null), scheme.at(1), budget, scheme.at(0).to_string());
+      gxc_token_tester::approve(scheme.at(1), gacha_account_name, budget);
+      produce_blocks(1);
+   }
+
+   action_result open(std::array<name,2> scheme, std::vector<grade> &grades, extended_asset budget, const string& expiration, fc::optional<uint8_t> precision, fc::optional<uint32_t> deadline_sec = fc::optional<uint32_t>()) {
+      return PUSH_ACTION(gacha_account_name, scheme.at(1), (scheme)(grades)(budget)(expiration)(precision)(deadline_sec));
+   }
+
+   action_result issue(account_name to, std::array<name,2> scheme, const string& dseedhash, fc::optional<uint64_t> id = fc::optional<uint64_t>()) {
+      return PUSH_ACTION(gacha_account_name, scheme.at(1), (to)(scheme)(dseedhash)(id));
+   }
+
+   action_result setoseed(uint64_t id, const string& oseed, account_name owner) {
+      return PUSH_ACTION(gacha_account_name, owner, (id)(oseed));
+   }
+
+   action_result setdseed(uint64_t id, const string& dseed) {
+      return PUSH_ACTION(gacha_account_name, gacha_account_name, (id)(dseed));
+   }
+
+   action_result winreward(account_name owner, uint64_t id, int64_t score, extended_asset value) {
+      return PUSH_ACTION(gacha_account_name, gacha_account_name, (owner)(id)(score)(value));
+   }
+
+   action_result raincheck(account_name owner, uint64_t id, int64_t score) {
+      return PUSH_ACTION(gacha_account_name, gacha_account_name, (owner)(id)(score));
+   }
+};
+
+BOOST_AUTO_TEST_SUITE(gxc_gacha_tests)
+
+BOOST_FIXTURE_TEST_CASE(open_tests, gxc_gacha_tester) try {
+
+   std::array<name, 2> schm = {N(gacha1), N(eun2ce)};
+   vector<grade> gd = {{.reward = asset::from_string("5000.00 ENC"), .score = 1, .limit = 0}};
+   string expiration = "2020-07-28T12:43:00";
+
+   subopen(schm, EA("5000.00 ENC@eun2ce"));
+   BOOST_REQUIRE_EQUAL(success(),
+         open(schm, gd, EA("5000.00 ENC@eun2ce"), expiration, 1)
```

Comment:
Instead of comparing `success()`, please check whether correct scheme object is created in table.